### PR TITLE
[G2M] Plotly - fix formatted value on hovertext for tooltip

### LINKED ISSUE #202: Plotly - remove unformatted value from tooltip
**Issue:**

- the tooltip includes both formatted & unformatted data value
Screen Shot 2022-10-12 at 4 55 08 PM

### PR DESCRIPTION
**Pull request by kc-leung**
fixes #202 

- Fixed decimals format value for tooltip on hover
- Added percentage check for tooltip on hover text display ( when the total sum value is 100, text value won't be display instead only percentage will be visible)

**BEFORE:** 

Screen Shot 2022-10-13 at 10 17 21 AM

**AFTER:**

Screen Shot 2022-10-13 at 10 05 28 AM

**UPDATE:** ( percentage check for tooltip )

https://user-images.githubusercontent.com/46459367/195700351-1f59ca46-cd93-4cab-bc98-2fb150acc351.mov

### CODE REVIEW

**Comment by kc-leung** [comment]
@Hamaad102 I added hovertext in for getObjectByType in typeConfig return so I have to add it into the plotly.test.js array for each chart? I am not passing the test check there

**Comment by geoerika** [comment]
@kc-leung a bunch of tests are failing

**Comment by Hamaad102** [comment]
@kc-leung you would have to do provide a value unfortunately, otherwise you could add a default value to the function?

**Comment by kc-leung** [comment]
> @kc-leung a bunch of tests are failing

yea I am asking @Hamaad102 to confirm me a question regarding the fail cuz I added a new value in the object return. It seems that I have to add into the test as well because in the object value check doesn't have that newly added value

**Comment by geoerika** [comment]
@kc-leung while we get the reviews in, could you please upgrade package to 0.8.1-alpha.1 and make a pre-release using this branch so I can test it through widget-studio & snoke. I am trying to get this in the Cox app today.

**Comment by geoerika** [comment]
@kc-leung the percentage value logic that you implemented for Pie chart can also appear in the Bar chart.

Also, I mentioned yesterday, we should not show the unformatted value if we don't have a case when the values amount to 100%.
Like in this Bar chart, the value appears twice in tooltip, we should only keep the formatted one in this case.
Screen Shot 2022-10-14 at 9 34 16 AM

File: src/components/plotly/shared/utils.js [hunk 1]
```diff
@@ -55,48 +55,60 @@ const getText = (value, formatting) => {
   if (formatting && typeof formatting === 'function') {
     return formatting(value)
   }
-  return value
+  return Number(value).toFixed(value % 1 && 2)
```
[the remark this answers is not in the document]
Comment (kc-leung) [reply src/components/plotly/shared/utils.js:58]: it is to round a number only if it has decimal places otherwise 55 will be rounded to 55.00

**Comment by Hamaad102** [comment]
@kc-leung are these changes only for the pie chart? I noticed on the plotly bar horizontal formatted that the decimal values are still unformatted.

Screenshot (23)

Otherwise, it all looks good!

**Comment by kc-leung** [comment]
> @kc-leung are these changes only for the pie chart? I noticed on the plotly bar horizontal formatted that the decimal values are still unformatted.
> 
> Screenshot (23)
> 
> Otherwise, it all looks good!

I only added the feature for Pie. I just had a talk with Erika to add it for all the other charts as well

**Review by woozyking — APPROVED** [review]

**Review by Hamaad102 — APPROVED** [review]

**Review by geoerika — APPROVED** [review]
You are still missing the end of file empty line in mock-data-pie-bar.json. Otherwise it looks good in dashboard. Let's update version to 0.8.1 merge & make a new full release.

**Pull request merged**